system/Model.php

```diff
@@ -2001,14 +1911,14 @@ public function __get(string $name)
 			return $this->{$name};
 		}
 
-		if (isset($this->db->$name))
```

Comment:
Do the `{` braces add something here? I never use them unless they are syntactically required.

Reply:
Same here. Maybe `{` make it more readable for people? Either way, we should remove them or keep them for all cases.

Reply:
Since we are accessing the properties of either `$this->db` and `$this->builder` dynamically using the value of `$name`, it is recommended that this dynamic access be made *explicit* by surrounding with curly braces. This helps to disambiguate usage by making it clear that the enclosed is dynamic. This apparently helps also in refactoring.

Also, migration from PHP 5 to PHP 7 has some changes in semantics in how variable variables are interpreted by the parser. This [guide](https://www.php.net/manual/en/migration70.incompatible.php#migration70.incompatible.variable-handling.indirect) shows the differences between PHP 5 and PHP 7 and how an explicit `{ }` can help in disambiguation.

`php-cs-fixer` rule: `explicit_indirect_variable`

Readings:
https://stackoverflow.com/questions/35598187/when-to-wrap-curly-braces-around-a-variable
https://www.php.net/manual/en/language.variables.variable.php

Reply:
I get the disambiguation, so I definitely agree with uses like below where you made this change:
`$result = $this->db->{$name}(...$params);`

But in cases where it is a simple variable property and there is no other way to interpret `$this->db->$name` it seems redundant and definitely not something I want to go through and apply everywhere in the framework nor in my code. If for some reason it helps static analysis or Rector or something I can get that, but if it is just to be consistent with the disambiguation cases then I would be in favor of dropping them.

Reply:
Ok, granted I would revert the changes, there are still places in the framework where this is applied. For example, old line 2011 of this `Model`, or in the constructor of `CodeIgniter\HTTP\ContentSecurityPolicy`. If these would be left out, then it would be inconsistent. Besides, you do not need to manually go over the framework to change these lines. You can use code style checkers like what I've said earlier, `php-cs-fixer` has this rule `explicit_indirect_variable` to make it consistent across the framework.

Reply:
It's true, we definitely aren't consistent. I say just leave it how it is and if we want to make a pass sometime down the road we can.

Overall this is a great move for Model! I love the clarity it brings to the code.